### PR TITLE
Allow any resume

### PR DESCRIPTION
**Pull request by amlcurran**
Adds a flag which allows a download to be resumed regardless of whether the server has returned an [Etag header](https://devcenter.heroku.com/articles/increasing-application-performance-with-http-cache-headers#content-based) or not

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/119/
Test FAILed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/120/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/Request.java [hunk 1]
```diff
@@ -223,6 +224,16 @@ public void allowScanningByMediaScanner() {
     }
 
     /**
+     * Always attempt to resume the download, regardless of whether the server returns
+     * a Etag header or not. **CAUTION** if the file has changed then this flag will
+     * result in undefined behaviour.
+     */
+    public Request alwaysAttemptResume() {
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/Request.java]: :+1: nice allowing the client to choose this and the jdoc

File: library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java [hunk 2]
```diff
@@ -356,6 +356,10 @@ private void clearHeaders() {
         requestHeaders.clear();
     }
 
+    public boolean isResumable() {
+        return alwaysResume || (eTag != null && isNoIntegrity());
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: :+1:

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java [hunk 3]
```diff
@@ -249,6 +249,12 @@ private DownloadContract() {
          */
         public static final String COLUMN_EXTRA_DATA = "extra_data";
 
+        /**
+         * The column used to flag whether a download should always attempt a resume
+         * if it is paused, regardless of what the server says
+         */
+        public static final String COLUMN_ALWAYS_RESUME = "alwaysResume";
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: Should we keep the column names consistent, `always_resume` ?

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: :+1:

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/123/
Test PASSed.

**Comment by Electryc** [comment]
Should we demo this usage in the pause/resume example in the `demo app` ?

**Comment by Electryc** [comment]
Just a little question about the demo, otherwise :+1: LGTM

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/126/
Test PASSed.

**Comment by Electryc** [comment]
any more :eyes: ?

**Comment by ronocod** [comment]
Second :+1:, time to :boat:

File: library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java (hunk 2 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: is `isNoIntegrity` used now outside the class? thinking of reducing visibility of the method

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: Yeah, I think so

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java#L774 (but that line could be pushed into the filedownloadinfo too tbh)

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: up2u

**Pull request merged**